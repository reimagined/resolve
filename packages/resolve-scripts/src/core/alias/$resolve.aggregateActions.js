export default () => {
  const exports = []

  exports.push(
    `import { createActions } from 'resolve-redux'`,
    `import aggregates from '$resolve.aggregates'`,
    ``,
    `const actions = {}`,
    `for(const aggregate of aggregates) {`,
    `  Object.assign(actions, createActions(aggregate))`,
    `}`,
    ``,
    `export default actions`
  )

  return {
    code: exports.join('\r\n')
  }
}
