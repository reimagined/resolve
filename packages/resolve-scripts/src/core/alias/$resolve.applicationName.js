export default ({ deployOptions }) => {
  const exports = []

  exports.push(
    `const applicationName = ${JSON.stringify(deployOptions.applicationName)}`,
    ``,
    `export default applicationName`
  )

  return {
    code: exports.join('\r\n')
  }
}
