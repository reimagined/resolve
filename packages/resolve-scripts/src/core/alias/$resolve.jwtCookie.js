import { envKey } from 'json-env-extract'

import { message } from '../constants'

export default ({ resolveConfig }) => {
  if (!resolveConfig.jwtCookie) {
    throw new Error(`${message.configNotContainSectionError}.jwtCookie`)
  }

  for (const optionsKey of Object.keys(resolveConfig.jwtCookie)) {
    if (resolveConfig.jwtCookie[optionsKey] in resolveConfig[envKey]) {
      throw new Error(`${message.clientEnvError}.jwtCookie.${optionsKey}`)
    }
  }

  const exports = []

  exports.push(
    `const jwtCookie = ${JSON.stringify(resolveConfig.jwtCookie, null, 2)}`,
    ``,
    `export default jwtCookie`
  )

  return {
    code: exports.join('\r\n')
  }
}
