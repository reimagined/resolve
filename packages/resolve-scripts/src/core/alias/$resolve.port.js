import { injectEnv, envKey } from 'json-env-extract'

import { message } from '../constants'

export default ({ resolveConfig, isClient }) => {
  if (isClient) {
    throw new Error(`${message.serverAliasInClientCodeError}$resolve.port`)
  }

  if (!resolveConfig.port) {
    throw new Error(`${message.configNotContainSectionError}.port`)
  }

  const config = {
    port: resolveConfig.port
  }
  Object.defineProperty(config, envKey, { value: resolveConfig[envKey] })

  const exports = []

  exports.push(
    `const config = ${injectEnv(config)}`,
    ``,
    `const port = config.port`,
    ``,
    `export default port`
  )

  return {
    code: exports.join('\r\n')
  }
}
