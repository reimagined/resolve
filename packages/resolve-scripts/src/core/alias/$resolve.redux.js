import { envKey } from 'json-env-extract'

import { message } from '../constants'

import resolveFile from '../resolve_file'

export default ({ resolveConfig }) => {
  if (!resolveConfig.redux) {
    throw new Error(`${message.configNotContainSectionError}.redux`)
  }

  if (resolveConfig.redux.store in resolveConfig[envKey]) {
    throw new Error(`${message.clientEnvError}.redux.store`)
  }
  const store = resolveFile(resolveConfig.redux.store)

  if (resolveConfig.redux.reducers in resolveConfig[envKey]) {
    throw new Error(`${message.clientEnvError}.redux.reducers`)
  }
  const reducers = resolveFile(resolveConfig.redux.reducers)

  if (resolveConfig.redux.middlewares in resolveConfig[envKey]) {
    throw new Error(`${message.clientEnvError}.redux.middlewares`)
  }
  const middlewares = resolveFile(resolveConfig.redux.middlewares)

  const exports = []

  exports.push(
    `import store from ${JSON.stringify(store)}`,
    `import reducers from ${JSON.stringify(reducers)}`,
    `import middlewares from ${JSON.stringify(middlewares)}`,
    ``,
    `export default {`,
    `  store, reducers, middlewares`,
    `}`
  )

  return {
    code: exports.join('\r\n')
  }
}
