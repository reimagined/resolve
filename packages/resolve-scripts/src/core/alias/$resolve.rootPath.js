import { envKey } from 'json-env-extract'

import { message } from '../constants'

export default ({ resolveConfig }) => {
  if (resolveConfig.rootPath === undefined) {
    throw new Error(`${message.configNotContainSectionError}.rootPath`)
  }

  if (resolveConfig.rootPath in resolveConfig[envKey]) {
    throw new Error(`${message.clientEnvError}.rootPath`)
  }
  const rootPath = resolveConfig.rootPath

  const exports = []

  exports.push(
    `const rootPath = ${JSON.stringify(rootPath)}`,
    ``,
    `export default rootPath`
  )

  return {
    code: exports.join('\r\n')
  }
}
