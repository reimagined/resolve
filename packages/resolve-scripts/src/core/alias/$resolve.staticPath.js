import { envKey } from 'json-env-extract'

import { message } from '../constants'

export default ({ resolveConfig, isClient }) => {
  if (isClient) {
    throw new Error(
      `${message.serverAliasInClientCodeError}$resolve.staticPath`
    )
  }

  if (resolveConfig.staticPath == null) {
    throw new Error(`${message.configNotContainSectionError}.staticPath`)
  }

  if (resolveConfig.staticPath in resolveConfig[envKey]) {
    throw new Error(`${message.clientEnvError}.staticPath`)
  }

  const exports = []

  exports.push(
    `const staticPath = ${JSON.stringify(resolveConfig.staticPath, null, 2)}`,
    ``,
    `export default staticPath`
  )

  return {
    code: exports.join('\r\n')
  }
}
