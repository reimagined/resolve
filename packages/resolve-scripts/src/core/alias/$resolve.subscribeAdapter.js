import { envKey } from 'json-env-extract'

import { message } from '../constants'

export default ({ resolveConfig }) => {
  if (!resolveConfig.subscribeAdapter) {
    throw new Error(`${message.configNotContainSectionError}.subscribeAdapter`)
  }

  if (resolveConfig.subscribeAdapter.module in resolveConfig[envKey]) {
    throw new Error(`${message.clientEnvError}.subscribeAdapter.module`)
  }
  for (const optionsKey of Object.keys(
    resolveConfig.subscribeAdapter.options
  )) {
    if (
      resolveConfig.subscribeAdapter.options[optionsKey] in
      resolveConfig[envKey]
    ) {
      throw new Error(
        `${message.clientEnvError}.subscribeAdapter.options.${optionsKey}`
      )
    }
  }

  const exports = []

  exports.push(
    `import module from ${JSON.stringify(
      resolveConfig.subscribeAdapter.module
    )}`,
    ``,
    `const options = ${JSON.stringify(
      resolveConfig.subscribeAdapter.options,
      null,
      2
    )}`,
    ``,
    `export default { module, options }`
  )

  return {
    code: exports.join('\r\n')
  }
}
