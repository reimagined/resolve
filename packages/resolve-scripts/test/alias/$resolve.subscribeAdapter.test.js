import path from 'path'
import { extractEnv } from 'json-env-extract'

import alias from '../../src/core/alias/$resolve.subscribeAdapter'
import normalizePaths from './normalize_paths'

describe('base config works correctly', () => {
  const resolveConfig = extractEnv(`
    {
      subscribeAdapter: {
        module: "${path.resolve(__dirname, 'files/testAdapter.js')}",
        options: {
          url: 'http://test.test'
        }
      }
    }
  `)

  test('[client]', () => {
    expect(
      normalizePaths(
        '\r\n' +
          alias({
            resolveConfig,
            isClient: true
          }).code +
          '\r\n'
      )
    ).toMatchSnapshot()
  })

  test('[server]', () => {
    expect(
      normalizePaths(
        '\r\n' +
          alias({
            resolveConfig,
            isClient: false
          }).code +
          '\r\n'
      )
    ).toMatchSnapshot()
  })
})

test('config with process.env failed', () => {
  const resolveConfig = extractEnv(`
    {
      subscribeAdapter: {
        module: process.env.SUBSCRIBE_ADAPTER,
        options: {
          url: process.env.SUBSCRIBE_OPTIONS_URL
        }
      }
    }
  `)

  expect(() =>
    normalizePaths(
      '\r\n' +
        alias({
          resolveConfig,
          isClient: false
        }).code +
        '\r\n'
    )
  ).toThrow()
})

test('config with process.env (v2) failed', () => {
  const resolveConfig = extractEnv(`
    {
      subscribeAdapter: {
        module: "${path.resolve(__dirname, 'files/testAdapter.js')}",
        options: {
          url: process.env.SUBSCRIBE_OPTIONS_URL
        }
      }
    }
  `)

  expect(() =>
    normalizePaths(
      '\r\n' +
        alias({
          resolveConfig,
          isClient: false
        }).code +
        '\r\n'
    )
  ).toThrow()
})
