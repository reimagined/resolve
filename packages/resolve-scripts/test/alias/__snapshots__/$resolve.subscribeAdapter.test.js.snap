// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base config works correctly [client] 1`] = `
"
import module from \\"<MONOREPO_DIR>/packages/resolve-scripts/test/alias/files/testAdapter.js\\"

const options = {
  \\"url\\": \\"http://test.test\\"
}

export default { module, options }
"
`;

exports[`base config works correctly [server] 1`] = `
"
import module from \\"<MONOREPO_DIR>/packages/resolve-scripts/test/alias/files/testAdapter.js\\"

const options = {
  \\"url\\": \\"http://test.test\\"
}

export default { module, options }
"
`;
